JetBrains: update JCEF when settings change

vdavid opened the pull request:
Fixes https://github.com/sourcegraph/sourcegraph/issues/36537

Notes:
- It uses the message bus.
- I've also extracted "getConfigAsJson" to ConfigUtil
- I've also made sure that the new listener connection is disposed of with the service.
- Made JavaToJSBridge pool use two queries, to make sure settings change can be applied at the same time as the theme change. It's because IF you change the theme from light to dark AND change plugin settings, then these need to be applied simultaneously when you press "OK" in Settings.

## Test plan

[2 min Loom demo](https://www.loom.com/share/3af288b1a656460c9c8697d56aebd373)

## App preview:

- [Web](https://sg-web-dv-jetbrains-update-jcef-when.onrender.com)
- [Storybook](https://5f0f381c0e50750022dc6bf7-wgzywerznm.chromatic.com)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

philipp-spiess reviewed (commented):
I’m not super fond on the way we handle the query pooling for multiple reasons:

- It already gets in our way I assume, otherwise you wouldn't have added support for a second query runner.
- It's inherently leaky: The call sites currently need to check if a runner is available and handle the case where it isn't (which in this case would no-op and prevent the updates from being forwarded to the web view).

I posted some idea on how we could improve that inline. This would probably be easy to add but given Java's big standard library I wouldn't be surprised if there's even something that we can already use (e.g. I found [BlockingQueue](https://docs.oracle.com/javase/7/docs/api/java/util/concurrent/BlockingQueue.html) which seems fitting :-)

vdavid commented:
Fixed this branch based on @philipp-spiess's review: the pooled query solution was ugly and the interface of the Java→JS bridge was inconvenient. I've solved it with a mutex lock and two futures, and now it's a lot more flexible and convenient solution. Thanks again for the great suggestions, @philipp-spiess!

Also found and fixed a bug: when the access token was modified without a restart, the EventSource polyfill was not updated with the new token. Now it's being updated.

Didn't create a new Loom because the behavior demonstrated there is the same, just in the background, the solution is nicer.

philipp-spiess commented on client/jetbrains/src/main/java/com/sourcegraph/browser/JavaToJSBridge.java:
```diff
@@ -1,60 +1,93 @@
                 return null;
             };
             query.addHandler(handler);
             browser.getCefBrowser().executeJavaScript(js, browser.getCefBrowser().getURL(), 0);
```
Should this also be inside the try/catch block so that we release the lock when it throws for some reason?

philipp-spiess reviewed (approved):
Awesomesauce! Thanks for the detailed comments they make the new logic very easy to grasp.
